linera-sdk/src/test/block.rs

```diff
@@ -158,73 +154,94 @@ impl BlockBuilder {
         self
     }
 
-    /// Receives an incoming message referenced by the [`MessageId`].
-    ///
-    /// The block that produces the message must have already been executed by the test validator,
-    /// so that the message is already in the inbox of the microchain this block belongs to.
-    pub fn with_incoming_bundle(&mut self, message_id: MessageId) -> &mut Self {
-        self.incoming_bundles
-            .push((message_id, MessageAction::Accept));
-        self
-    }
-
-    /// Rejects an incoming message referenced by the [`MessageId`].
+    /// Receives incoming message bundles by specifying them directly.
     ///
-    /// The block that produces the message must have already been executed by the test validator,
-    /// so that the message is already in the inbox of the microchain this block belongs to.
-    pub fn with_message_rejection(&mut self, message_id: MessageId) -> &mut Self {
-        self.incoming_bundles
-            .push((message_id, MessageAction::Reject));
+    /// This is an internal method that bypasses the check to see if the messages are already
+    /// present in the inboxes of the microchain that owns this block.
```

Comment:
Would it make sense to allow receiving individual bundles, instead of all relevant bundles from a certificate?

Reply:
This would probably be code that has to be rewritten immediately with https://github.com/linera-io/linera-protocol/pull/2343, so I'd prefer to do that later, or when we actually need it.

Reply:
Nice, makes sense :+1: